lib/capistrano/net_storage/utils.rb

```diff
@@ -18,28 +18,34 @@ def config
       # You can provide either of +dest_dir+ or +dest_path+, or files are copied to same pathes
       def upload_files(files, dest_dir: nil, dest_path: nil)
         c = config
-        files.each do |src|
-          basename = File.basename(src)
-          dest = dest_path || begin
-            dir = dest_dir || File.dirname(src)
-            File.join(dir, basename)
-          end
-
-          if c.upload_files_by_rsync?
-            Parallel.each(c.servers, in_threads: c.max_parallels) do |host|
+        if c.upload_files_by_rsync?
+          on c.servers, in: :groups, limit: c.max_parallels do |host|
```

Comment:
I am not sure if it is possible or not to use `run_locally` inside `on` block.

Reply:
`run_locally` function correctly inside `on` block.

Reply:
Okey.